bin-src/init.ts

```diff
@@ -0,0 +1,144 @@
+import meow from 'meow';
+import prompts from 'prompts';
+import type { PackageJson } from 'read-pkg-up'
+import readPkgUp from 'read-pkg-up';
+import { getCliCommand, parseNi } from '@antfu/ni';
+
+import noPackageJson from '../node-src/ui/messages/errors/noPackageJson';
+
+export const TestFramework = {
+    STORYBOOK: 'storybook',
+    PLAYWRIGHT: 'playwright',
+    CYPRESS: 'cypress'
+};
+type TestFrameworkType = typeof TestFramework[keyof typeof TestFramework];
+
+export const getPackageManagerInstallCommand = async (args: string[]) => {
+    return await getCliCommand(parseNi, args, { programmatic: true });
+};
+
+const getSBConfigFilePath = async () => {
+    // Walks up directory tree to find nearest Storybook config file.
+    return await findUp(['.storybook/main.ts', '.storybook/main.js', '.storybook/main.tsx', '.storybook/main.jsx', '.storybook/main.mjs', '.storybook/main.cjs'])
+}
+
+export const addChromaticScriptToPackageJson = async ({ packageJson, packagePath }) => {
+    try {
+        const json = {
+            ...packageJson,
+            scripts: {
+                ...packageJson?.scripts,
+                chromatic: `npx chromatic`
+            }
+        }
+        await writeFile(packagePath, json, { spaces: 2 });
+    } catch (e) {
+        console.warn(e)
+    }
+}
+
+export const createChromaticConfigFile = async ({configFile, buildScriptName = null}) => {
+    await writeFile(configFile, {
+        autoAcceptChanges: "main",
+        ...(buildScriptName && {
+            buildScriptName
+        }),
+        onlyChanged: true,
+        skip: "dependabot/**"
+    });
+}
+
+export const installArchiveDependencies = async (packageJson: PackageJson, testFramework: TestFrameworkType) => {
+    let installArgs = ['-D', 'chromatic',`chromatic-${testFramework}`, 'storybook@next', '@storybook/addon-essentials@next', '@storybook/server-webpack5@next']
+    const sbConfigPath = await getSBConfigFilePath()
+    const sbVersion = packageJson?.devDependencies?.storybook || packageJson?.dependencies?.storybook
+    if(sbConfigPath && sbVersion) {
+        installArgs = ['-D', 'chromatic',`chromatic-${testFramework}`, `@storybook/server-webpack5@${sbVersion}`]
+    }
+    const installCommand = await getPackageManagerInstallCommand(installArgs)
+    await execaCommand(installCommand)
+}
+
+const intializeChromatic = async ({testFramework, packageJson, packagePath}: {testFramework: TestFrameworkType, packageJson: PackageJson, packagePath: string }) => {
+    await addChromaticScriptToPackageJson({ packageJson, packagePath });
+    await createChromaticConfigFile({configFile: 'chromatic.config.json'})
+    switch (testFramework) {
+        case TestFramework.CYPRESS:
+            await installArchiveDependencies(packageJson, TestFramework.CYPRESS)
+            break;
+        case TestFramework.PLAYWRIGHT:
+            await installArchiveDependencies(packageJson, TestFramework.PLAYWRIGHT)
+            break;
```

Comment:
```suggestion
        case TestFramework.CYPRESS:
        case TestFramework.PLAYWRIGHT:
            await installArchiveDependencies(packageJson, TestFramework.PLAYWRIGHT)
            break;
```

Reply:
installArchiveDependencies takes in a test framework that determines which dependency will get installed. So, that's why they are separate cases. Am I missing something from this suggestion?

Reply:
Oh, yeah right, but you could pass `testFramework`, but really its just a qn of style..